scripts/flash_algo.py

```diff
@@ -0,0 +1,363 @@
+    algo_data = bytearray(algo_size)
+    for section in (sect_ro, sect_rw):
+        start = section["sh_addr"]
+        size = sect_ro["sh_size"]
+        algo_data[start:start + size] = section.data()
+    return algo_data
+
+
+class PackFlashInfo(object):
+    """Wrapper class for the non-executable information in an FLM file"""
+
+    FLASH_DEVICE_STRUCT = "<H128sHLLLLBxxxLL"
+    FLASH_SECTORS_STRUCT = "<LL"
+    SECTOR_END = 0xFFFFFFFF
+
+    def __init__(self, elf_simple):
+        dev_info = elf_simple.symbols["FlashDevice"]
+        info_start = dev_info.value
+        info_size = struct.calcsize(self.FLASH_DEVICE_STRUCT)
+        data = elf_simple.read(info_start, info_size)
+        values = struct.unpack(self.FLASH_DEVICE_STRUCT, data)
+
+        self.version = values[0]
+        self.name = values[1].strip("\x00")
+        self.type = values[2]
+        self.start = values[3]
+        self.size = values[4]
+        self.page_size = values[5]
+        self.value_empty = values[7]
+        self.prog_timeout_ms = values[8]
+        self.erase_timeout_ms = values[9]
+
+        sector_entry_size = struct.calcsize(self.FLASH_SECTORS_STRUCT)
+        index = 0
+        sector_size, sector_start = 0, 0
+        sector_info_list = []
+        while True:
+            sector_info_start = (info_start + info_size +
+                                 index * sector_entry_size)
+            data = elf_simple.read(sector_info_start, sector_entry_size)
+            sector_size, sector_start = struct.unpack(self.FLASH_SECTORS_STRUCT, data)
+            if (sector_size == self.SECTOR_END and
+               sector_start == self.SECTOR_END):
+                break
+            sector_info_list.append((sector_start, sector_size))
+            index += 1
+        self.sector_info_list = sector_info_list
+
+    def __str__(self):
+        desc = ""
+        desc += "Flash Device:" + os.linesep
+        desc += "  name=%s" % self.name + os.linesep
+        desc += "  version=0x%x" % self.version + os.linesep
+        desc += "  type=%i" % self.type + os.linesep
+        desc += "  start=0x%x" % self.start + os.linesep
+        desc += "  size=0x%x" % self.size + os.linesep
+        desc += "  page_size=0x%x" % self.page_size + os.linesep
+        desc += "  value_empty=0x%x" % self.value_empty + os.linesep
+        desc += "  prog_timeout_ms=%i" % self.prog_timeout_ms + os.linesep
+        desc += "  erase_timeout_ms=%i" % self.erase_timeout_ms + os.linesep
+        desc += "  sectors:" + os.linesep
+        for sector_start, sector_size in self.sector_info_list:
+            desc += ("    start=0x%x, size=0x%x" %
+                     (sector_start, sector_size) + os.linesep)
+        return desc
+
+
+class SymbolSimple(object):
+    """Wrapper for symbol object"""
+
+    def __init__(self, name, value, size):
+        self.name = name
+        self.value = value
+        self.size = size
+
+
+class ElfFileSimple(object):
```

Comment:
If this class is a wrapper, why does it not inherit from the thing that it wraps?